File: internal/indexer/goldeniir/testdata/typescript/failures.ts
// Error("msg") yields a constructed failure (the message); a custom error class
// without a message yields a sentinel (the class name); a re-throw yields a
// propagated failure (the forwarded identifier).
class NotFoundError extends Error {}

export function loadUser(id: string): string {
  if (id === "") {
    throw new Error("empty_id")
  }
  if (id === "missing") {
    throw new NotFoundError()
  }
  return id
}

export function retry(fn: () => void): void {
  try {
    fn()
  } catch (err) {
    throw err // re-throw — a propagated failure (source: err)
  }
}

File: internal/indexer/goldeniir/testdata/python/failures.py
class NotFoundError(Exception):
    pass


def load_user(user_id):
    # ValueError("msg") -> constructed (message); a bare custom class / call with
    # no message -> sentinel (exception type); a bare `raise` re-raise ->
    # propagated.
    if user_id == "":
        raise ValueError("empty_id")
    if user_id == "missing":
        raise NotFoundError()
    if user_id == "closed":
        raise NotFoundError
    return user_id


def retry(fn):
    try:
        fn()
    except Exception:
        raise  # re-raise — a propagated failure (anonymous, code "propagated")
